src/funcs.c

```diff
@@ -791,9 +645,10 @@ Obj DoPartialUnWrapFunc(Obj func, Obj args)
     ASS_LVAR(named+1, args);
 
     /* execute the statement sequence                                      */
-    REMEMBER_LOCKSTACK();
```

Comment:
The `REMEMBER_LOCKSTACK` above previously did nothing, as this function (`DoPartialUnWrapFunc`) never was called in HPC-GAP; this is the bug fixed by this PR. Since we now call `DoPartialUnWrapFunc` from HPC-GAP, we must  make sure to call `REMEMBER_LOCKSTACK` *before* we take any locks. Also, we call `CHECK_RECURSION_BEFORE` as the very first thing, before any other argument handling.